engine/docker/convert.go

```diff
@@ -18,14 +20,28 @@ func toConfig(proc *engine.Step) *container.Config {
 		AttachStdout: true,
 		AttachStderr: true,
 	}
+
+	if len(proc.Networks) != 0 {
+		ports := make(nat.PortSet)
+		for _, network := range proc.Networks {
+			for _, port := range network.Ports {
+				natPort, err := nat.NewPort("TCP", fmt.Sprintf("%d", port))
+				if err != nil {
+					panic(err)
+				}
+				ports[natPort] = struct{}{}
+			}
+		}
+		config.ExposedPorts = ports
+	}
```

Comment:
can you explain more what this does? The current implementation allows containers to communicate using hostname aliases, and does not require any additional port configuration to work properly:

```
		for _, net := range proc.Networks {
			err = e.client.NetworkConnect(ctx, net.Name, proc.Name, &network.EndpointSettings{
				Aliases: net.Aliases,
			})
			if err != nil {
				return err
			}
		}
```